communicator/vmturbocommunicator/server_message_handler.go

```diff
@@ -268,17 +278,45 @@ func (handler *MesosServerMessageHandler) HandleAction(serverMsg *comm.Mediation
 func (handler *MesosServerMessageHandler) NewMesosProbe(previousUseMap map[string]*util.CalculatedUse) (*util.MesosAPIResponse, error) {
 	fullUrl := "http://" + handler.meta.MesosIP + ":" + handler.meta.MesosPort + "/state"
 	glog.V(4).Infof("The full Url is ", fullUrl)
+
 	req, err := http.NewRequest("GET", fullUrl, nil)
 
+	req.Header.Add("content-type", "application/json")
+
+	if handler.meta.DCOS {
+		req.Header.Add("authorization", "token="+handler.meta.Token)
+	}
+
 	glog.V(4).Infof("%+v", req)
 	client := &http.Client{}
 	resp, err := client.Do(req)
+
 	if err != nil {
-		glog.Errorf("Error getting response: %s \n", err)
+		glog.V(3).Infof("Error in GET request to mesos master: %s\n", err)
 		return nil, err
 	}
-	defer resp.Body.Close()
+	// Get token if response if OK
+	if resp.Status == "" {
+		glog.V(3).Infof("Empty response status\n")
+		return nil, err
+	}
+
+	statusmsg := strings.Split(resp.Status, " ")
 
+	if statusmsg[0] != "200" && statusmsg[1] == "Unauthorized" {
```

Comment:
for now let's assume if the GET failed, it's because of token expiration. (Change of username and password is a relatively less frequent operation). So the plan is to have a call and install new mesosturbo and turbonomic tomorrow, and have a follow up call next week so that we can see if the token expiration handler is working.